redfish-core/lib/assembly.hpp

```diff
@@ -711,6 +712,224 @@ inline void getChassis(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+        messages::internalError(aResp->res);
+        return;
+    }
+
+    using associationList =
+        std::vector<std::tuple<std::string, std::string, std::string>>;
+
+    crow::connections::systemBus->async_method_call(
+        [aResp, assemblyUriPropPath, assemblyParentObjPath, assembledObjPath,
+         assembledUriVal](const boost::system::error_code ec,
+                          const std::variant<associationList>& associations) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR
+                    << "DBUS response error [" << ec.value() << " : "
+                    << ec.message()
+                    << "] when tried to get the Associations from ["
+                    << assemblyParentObjPath.str
+                    << "] to fill Assembly id of the assembled object ["
+                    << assembledObjPath.str << "]";
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            const associationList* value =
+                std::get_if<associationList>(&associations);
+            if (value == nullptr)
+            {
+                BMCWEB_LOG_ERROR
+                    << "Failed to get the Associations from ["
+                    << assemblyParentObjPath.str
+                    << "] to fill Assembly id of the assembled object ["
+                    << assembledObjPath.str << "]";
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            std::vector<std::string> assemblyAssoc;
+            for (const auto& association : *value)
+            {
+                if (std::get<0>(association) != "assembly")
+                {
+                    continue;
+                }
+                assemblyAssoc.emplace_back(std::get<2>(association));
+            }
+
+            if (assemblyAssoc.empty())
+            {
+                BMCWEB_LOG_ERROR
+                    << "No assembly associations in the ["
+                    << assemblyParentObjPath.str
+                    << "] to fill Assembly id of the assembled object ["
+                    << assembledObjPath.str << "]";
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            // Mak sure whether the retrieved assembly associations are
+            // implemented before finding the assembly id as per bmcweb
+            // Assembly design.
+            crow::connections::systemBus->async_method_call(
+                [aResp, assemblyUriPropPath, assemblyParentObjPath,
+                 assembledObjPath, assemblyAssoc, assembledUriVal](
+                    const boost::system::error_code ec,
+                    const std::vector<
+                        std::pair<std::string,
+                                  std::vector<std::pair<
+                                      std::string, std::vector<std::string>>>>>&
+                        subtrees) {
```

Comment:
I have never seen it called "subtrees", just subtree and then the member are object. e.g. 
https://github.com/openbmc/bmcweb/blob/55f79e6fe156505cdaddd521212a29b2a977193a/redfish-core/lib/systems.hpp#L263